tests/test_blank_footprint.py

```diff
@@ -0,0 +1,218 @@
+        check_all_dependencies=False
+    )
+
+    if data is not None:
+        sim.modules['HealthSystem'].parameters['availability_estimates'] = data
+
+    sim.make_initial_population(n=100)
+
+    if run:
+        sim.simulate(end_date=start_date)
+
+    return sim
+
+
+def get_dummy_hsi_event_instance(module, facility_id=None, blank = False):
+    """Make an HSI Event that runs for person_id=0 in a particular facility_id and requests consumables,
+    and for which its parent is the identified module."""
+
+    _facility_level = find_level_of_facility_id(facility_id)
+
+    class HSI_Dummy_no_blank(HSI_Event, IndividualScopeEventMixin):
+        def __init__(self, module, person_id):
+            super().__init__(module, person_id=person_id)
+            self.TREATMENT_ID = 'Dummy'
+            self.ACCEPTED_FACILITY_LEVEL = _facility_level
+            self._facility_id = facility_id
+            self.blank = blank_footprint
+            if self.blank:
+                self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({})
+            else:
+                self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint({'ConWithDCSA': 1})
+
+
+
+        def apply(self, person_id, squeeze_factor):
+            if squeeze_factor != np.inf:
+                # Check that this appointment is being run and run not with a squeeze_factor that signifies that a cadre
+                # is not at all available.
+                self.did_run = True
+
+
+    hsi_dummy = HSI_Dummy_no_blank(module=module, person_id=0)
+    hsi_dummy.initialise()
+    hsi_dummy.facility_info = module.sim.modules['HealthSystem']._facility_by_facility_id[facility_id]
+    return hsi_dummy
+
+
+def get_dummy_hsi_event_instance_blank_footprint(module, facility_id=None):
+    """Make an HSI Event that runs for person_id=0 in a particular facility_id and requests consumables,
+    and for which its parent is the identified module."""
+
+    _facility_level = find_level_of_facility_id(facility_id)
+
+    class HSI_Dummy_with_blank(HSI_Event, IndividualScopeEventMixin):
+        def __init__(self, module, person_id):
+            super().__init__(module, person_id=person_id)
+            self.TREATMENT_ID = 'Dummy'
+            self.ACCEPTED_FACILITY_LEVEL = _facility_level
+            self.EXPECTED_APPT_FOOTPRINT = self.make_blank_appt_footprint()
+            self._facility_id = facility_id
+
+        def apply(self, person_id, squeeze_factor):
+            if squeeze_factor != np.inf:
+                # Check that this appointment is being run and run not with a squeeze_factor that signifies that a cadre
+                # is not at all available.
+                self.did_run = True
+        def make_blank_appt_footprint(self):
+            return {}
+
+    hsi_dummy = HSI_Dummy_with_blank(module=module, person_id=0)
+    hsi_dummy.initialise()
+    hsi_dummy.facility_info = module.sim.modules['HealthSystem']._facility_by_facility_id[facility_id]
+    return hsi_dummy
+
```

Comment:
We can introduce a new function here to set the person and district since this seems to be called repeatedly at least in 2 spots. More on this in the comment below. Please note that you may change the function name as you see fit. However, the function can be as follows:
```suggestion
def set_person_district_id(sim):
      # sets facility and person to 0
      sim.population.props.at[0, 'district_of_residence'] = mfl.set_index('Facility_ID').loc[0].District
      return sim
```